### PR TITLE
[Dygraph] yolov3 eval success

### PR DESCRIPTION
**Pull request by wangxinxin08**

### CODE REVIEW

File: tools/train.py [hunk 1]
```diff
@@ -46,7 +31,7 @@
 def parse_args():
     parser = ArgsParser()
     parser.add_argument(
-        "--weight_type",
+        "-ckpt_type",
```
Comment (jerrywgz) [inline tools/train.py:34]: 这里的save_load用新合入的代码吧

File: tools/train.py [hunk 2]
```diff
@@ -122,41 +107,41 @@ def run(FLAGS, cfg, place):
...
     if ParallelEnv().nranks > 1:
         model = paddle.DataParallel(model)
 
     # Run Train
-    start_iter = 0
     time_stat = deque(maxlen=cfg.log_iter)
     start_time = time.time()
     end_time = time.time()
-    # Run Train
-    start_epoch = optimizer.state_dict()['LR_Scheduler']['last_epoch']
```
Comment (jerrywgz) [inline tools/train.py:150]: 这里也保留start_epoch和后面cur_eid的计算逻辑吧，为了支持恢复训练

File: tools/train.py (hunk 1 above)
Comment (wangxinxin08) [reply tools/train.py:34]: Done, Thanks

File: tools/train.py (hunk 2 above)
Comment (wangxinxin08) [reply tools/train.py:150]: Done, Thanks

File: configs/_base_/readers/yolov3_reader.yml [hunk 3]
```diff
@@ -44,8 +44,8 @@ TestReader:
     image_shape: [3, 608, 608]
     fields: ['image', 'im_size', 'im_id']
```
Comment (jerrywgz) [inline configs/_base_/readers/yolov3_reader.yml:45]: change to im_shape, scale_factor

File: ppdet/data/source/dataset.py [hunk 4]
```diff
@@ -39,13 +39,31 @@ def __init__(self,
         self.image_dir = image_dir if image_dir is not None else ''
         self.sample_num = sample_num
         self.use_default_label = use_default_label
+        self.epoch = 0
+        self.mixup_epoch = kwargs.get('mixup_epoch', -1)
+        self.mixup = kwargs.get('mixup', False) or self.mixup_epoch > -1
```
Comment (qingqing01) [inline ppdet/data/source/dataset.py:44]: mixup_epoch和mixup两个参数可以简化一个吧

File: ppdet/data/source/dataset.py [hunk 5]
```diff
@@ -39,13 +39,31 @@ def __init__(self,
...
+        self.cutmix_epoch = kwargs.get('cutmix_epoch', -1)
+        self.cutmix = kwargs.get('cutmix', False) or self.cutmix_epoch > -1
+        self.mosaic_epoch = kwargs.get('mosaic_epoch', -1)
+        self.mosaic = kwargs.get('mosaic', False) or self.mosaic_epoch > -1
```
Comment (qingqing01) [inline ppdet/data/source/dataset.py:48]: 同上

File: ppdet/modeling/loss/iou_loss.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+class IouLoss(object):
+    """
+    iou loss, see https://arxiv.org/abs/1908.03851
+    loss = 1.0 - iou * iou
+    Args:
+        loss_weight (float): iou loss weight, default is 2.5
+        max_height (int): max height of input to support random shape input
+        max_width (int): max width of input to support random shape input
+        ciou_term (bool): whether to add ciou_term
+        loss_square (bool): whether to square the iou term
```
Comment (qingqing01) [inline ppdet/modeling/loss/iou_loss.py:38]: 同上

File: ppdet/data/source/dataset.py (hunk 4 above)
Comment (wangxinxin08) [reply ppdet/data/source/dataset.py]: Done, Thanks

File: ppdet/data/source/dataset.py (hunk 5 above)
Comment (wangxinxin08) [reply ppdet/data/source/dataset.py]: Done, Thanks

File: ppdet/modeling/loss/iou_loss.py (hunk 6 above)
Comment (wangxinxin08) [reply ppdet/modeling/loss/iou_loss.py:38]: Done, Thanks

**Review by heavengate — APPROVED** [review]
